File: Programming Basics/C++/Nested loops/SumPrimeNonPrime/SumPrimeNonPrime.cpp
#include <iostream>
#include <string>
#include <math.h>
using namespace std;

int main()
{
    int primeSum = 0, nonPrimeSum = 0;

    string command;
    cin >> command;
    while (command != "stop")
    {
        int number = stoi(command);

        if (number < 0) cout << "Number is negative." << endl;
        else
        {
            bool isPrime = true;
            double numSqrt = sqrt(number);
            for (int i = 2; i <= numSqrt; i++)
            {
                if (number % i == 0)
                {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) primeSum += number;
            else nonPrimeSum += number;
        }

        cin >> command;
    }

    cout << "Sum of all prime numbers is: " << primeSum << endl;
    cout << "Sum of all non prime numbers is: " << nonPrimeSum << endl;
}

File: Programming Basics/JavaScript/Nested loops/Exercises/sum-prime-non-prime.js
function solve(input) {
    let primeSum = 0;
    let nonPrimeSum = 0;

    let index = 0;
    while (input[index] !== "stop") {
        const number = Number(input[index]);

        if (number < 0) console.log("Number is negative.");
        else {
            let isPrime = true;
            let numSqrt = Math.sqrt(number);
            for (let i = 2; i <= numSqrt; i++) {
                if (number % i === 0) {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) primeSum += number;
            else nonPrimeSum += number;
        }

        index++;
    }

    console.log(`Sum of all prime numbers is: ${primeSum}`);
    console.log(`Sum of all non prime numbers is: ${nonPrimeSum}`);
}
